flink-runtime/src/main/java/org/apache/flink/streaming/api/operators/InternalTimeServiceManagerImpl.java

```diff
@@ -156,6 +159,15 @@ public <N> InternalTimerService<N> getInternalTimerService(
         return timerService;
     }
 
+    <N> InternalTimerServiceImpl<K, N> restoreTimeService(
+            String name, TimerSerializer<K, N> timerSerializer) {
+        if (priorityQueueSetFactory instanceof AsyncKeyedStateBackend) {
```

Comment:
How about adding `isAsyncState` as a member of this manager? Add pass this in via `create`